### PR TITLE
Use git-version-gen for helper image version

### PR DESCRIPTION
**Pull request by bdevcich**

### CODE REVIEW

**Review by ajfloeder — APPROVED** [review]

File: helper_image/Makefile [hunk 1]
```diff
@@ -9,16 +8,19 @@ IMG ?= ghcr.io/nearnodeflash/nnf-integration-test-helper:master
...
+docker-build: VERSION ?= $(shell cat ../.version)
+docker-build: .version ## Build docker image with the manager.
+	$(CONTAINER_TOOL) build -t ${IMG}:${VERSION} .
 
 .PHONY: docker-push
-docker-push: ## Push docker image with the manager.
-	$(CONTAINER_TOOL) _push ${IMG}
+docker-push: VERSION ?= $(shell cat ../.version)
+docker-push: .version ## Push docker image with the manager.
+	$(CONTAINER_TOOL) _push ${IMG}:${VERSION}
```
Comment (roehrich-hpe) [inline helper_image/Makefile:18]: underscore-push?

**Review by matthew-richerson — APPROVED** [review]

File: helper_image/Makefile [hunk 2]
```diff
@@ -28,11 +30,18 @@ kind-push: ## Push docker image to kind
...
-	- $(CONTAINER_TOOL) buildx build --push --platform=$(PLATFORMS) --tag ${IMG} -f Dockerfile.cross .
+	- $(CONTAINER_TOOL) buildx build --push --platform=$(PLATFORMS) --tag ${IMG}:${VERSION} -f Dockerfile.cross .
 	- $(CONTAINER_TOOL) buildx rm project-v3-builder
 	rm Dockerfile.cross
+
+# Let .version be phony so that a git update to the workarea can be reflected
+# in it each time it's needed.
+.PHONY: .version
+.version: ## Uses the git-version-gen script to generate a tag version
+	../git-version-gen --fallback `git rev-parse HEAD` > ../.version
```
Comment (roehrich-hpe) [inline helper_image/Makefile:47]: I wonder if this should instead be:
  `make -C .. .version`

**Review by roehrich-hpe — APPROVED** [review]

File: helper_image/Makefile (hunk 1 above)
Comment (bdevcich) [reply helper_image/Makefile:18]: Not sure what happened there

File: helper_image/Makefile (hunk 2 above)
Comment (bdevcich) [reply helper_image/Makefile]: Done

**Review by roehrich-hpe — APPROVED** [review]

**Pull request merged**